src/contexts/SiteLaunchContext.tsx

```diff
@@ -0,0 +1,86 @@
+import { createContext, useContext, useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+
+import { useGetSiteLaunchStatus } from "hooks/siteDashboardHooks"
+
+import {
+  SiteLaunchDto,
+  SiteLaunchFrontEndStatus,
+  SiteLaunchStatusProps,
+  SITE_LAUNCH_TASKS_LENGTH,
+} from "types/siteLaunch"
+
+interface SiteLaunchContextProps {
+  siteLaunchStatusProps: SiteLaunchStatusProps
+  setSiteLaunchStatusProps: (
+    siteLaunchStatusProps: SiteLaunchStatusProps
+  ) => void
+}
+
+const SiteLaunchContext = createContext<SiteLaunchContextProps | null>(null)
+
+export const useSiteLaunchContext = (): SiteLaunchContextProps => {
+  const SiteLaunchContextData = useContext(SiteLaunchContext)
+  if (!SiteLaunchContextData)
+    throw new Error("useSiteLaunchContext must be used within an RoleProvider")
+  return SiteLaunchContextData
+}
+
+export const SiteLaunchProvider = ({
+  children,
+  initialSiteLaunchStatus = "NOT_LAUNCHED",
+  initialStepNumber = 0,
+}: {
+  children: React.ReactNode
+  initialSiteLaunchStatus?: SiteLaunchFrontEndStatus
+  initialStepNumber?: number
```

Comment:
nit: should prefer to have these as a separate type

Reply:
done